src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp

```diff
@@ -741,56 +743,99 @@ void WaveTrackMenuTable::OnMergeStereo(wxCommandEvent &)
    AudacityProject *const project = &mpData->project;
    auto &tracks = TrackList::Get( *project );
 
-   WaveTrack *const pTrack = static_cast<WaveTrack*>(mpData->pTrack);
-   wxASSERT(pTrack);
+   const auto first = tracks.Any<WaveTrack>().find(mpData->pTrack);
+   const auto left = *first;
+   const auto right = *std::next(first);
 
-   auto partner =
-      static_cast<WaveTrack*>(*tracks.Find(pTrack).advance(1));
-
-   if (pTrack->GetRate() != partner->GetRate()) {
-      using namespace BasicUI;
-      ShowMessageBox(XO(
-"Mono tracks must have the same sample rate in order to be combined into a "
-"stereo track"),
-         MessageBoxOptions{}
-           .Caption(XO("Error"))
-           .IconStyle(Icon::Error));
-      return;
-   }
-   if(pTrack->GetSampleFormat() != partner->GetSampleFormat())
+   const auto checkAligned = [](const WaveTrack& left, const WaveTrack& right)
    {
-      BasicUI::ShowMessageBox(XO(
-"Mono tracks must have the same sample format in order to be combined "
-"into a stereo track"),
+      const auto isPlainInterval = [](const WaveTrack::Interval& interval)
+      {
+         return interval.GetTrimLeft() == .0 &&
+            interval.GetTrimRight() == .0 &&
+            interval.GetStretchRatio() == 1.0;
+      };
+      const auto rightIntervals = right.Intervals();
+      for(const auto& a : left.Intervals())
+      {
+         if(!isPlainInterval(*a))
+            return false;
+
+         auto it = std::find_if(
+            rightIntervals.begin(),
+            rightIntervals.end(),
+            [&](const auto& b)
+            {
+               return a->Start() == b->Start() &&
+                  a->End() == b->End() &&
+                  isPlainInterval(*b);
+            });
+         if(it == rightIntervals.end())
+            return false;
+      }
+      return true;
+   };
```

Comment:
This big lambda that decides whether to query before proceeding might need reuse elsewhere, when we fix opening of 3.3 files or importing of 2.x files.

It might move to the recently added WaveTrackUtilities.cpp